### ISSUE TITLE
Incorrect "error : statements are not aligned" error being thrown

### ISSUE THREAD

**Issue by MichaelTontchev** [palantir/tslint#3065]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
    componentWillReceiveProps(nextProps: MyType) {
        if (this.props.someValue !== nextProps.someValue) {
            this.setState((previousState, props) => {
                return {
                    ...someVariable,
                    another: false,
                    items: {},
                }
            });
        }
    }
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "align": [
            true,
            "parameters",
            "statements"
        ],
        "ban": false,
        "class-name": false,
        "comment-format": [
            false,
            "check-space",
            "check-lowercase"
        ],
        "curly": false,
        "eofline": false,
        "forin": false,
        "indent": [ true, 4 ],
        "interface-name": false,
        "jsdoc-format": false,
        "label-position": true,
        "label-undefined": true,
        "max-line-length": [ false, 140 ],
        "member-ordering": [
            false,
            "public-before-private",
            "static-before-instance",
            "variables-before-functions"
        ],
        "no-any": false,
        "no-arg": false,
        "no-bitwise": false,
        "no-console": [
            true,
            "debug",
            "info",
            "time",
            "timeEnd",
            "trace"
        ],
        "no-consecutive-blank-lines": true,
        "no-construct": true,
        "no-constructor-vars": false,
        "no-debugger": false,
        "no-duplicate-key": true,
        "no-duplicate-variable": false,
        "no-empty": false,
        "no-eval": false,
        "no-string-literal": false,
        "no-switch-case-fall-through": false,
        "no-trailing-whitespace": false,
        "no-unreachable": true,
        "no-unused-expression": false,
        "no-unused-variable": false,
        "no-use-before-declare": false,
        "no-var-requires": false,
        "one-line": [
            false,
            "check-open-brace",
            "check-catch",
            "check-else",
            "check-whitespace"
        ],
        "quotemark": [ false, "double" ],
        "radix": false,
        "semicolon": false,
        "switch-default": false,
        "triple-equals": [ false, "allow-null-check" ],
        "typedef": [
            false,
            "call-signature",
            "parameter",
            "property-declaration",
            "variable-declaration",
            "member-variable-declaration"
        ],
        "typedef-whitespace": [
            false,
            {
                "call-signature": "nospace",
                "index-signature": "nospace",
                "parameter": "nospace",
                "property-declaration": "nospace",
                "variable-declaration": "nospace"
            }
        ],
        "use-strict": [
            false,
            "check-module",
            "check-function"
        ],
        "variable-name": false,
        "whitespace": [
            false,
            "check-branch",
            "check-decl",
            "check-operator",
            "check-separator",
            "check-type"
        ]
    }
}

```

#### Actual behavior
"error : statements are not aligned", with line and column pointing to the semicolon at the end of setState

#### Expected behavior
No errors

The workaround seems to be removing the semicolon it doesn't like (but which is there legally and in fact required if we set semicolons to be enforced strictly).

**Comment by MichaelTontchev** [comment]
Possibly relevant: https://github.com/palantir/tslint/issues/2608

**Comment by MichaelTontchev** [comment]
Ping on this...

**Comment by JoshuaKGoldberg** [comment]
👋 @MichaelTontchev, I can't reproduce this with TSLint 5.11. Looks like #2668 resolved this!
